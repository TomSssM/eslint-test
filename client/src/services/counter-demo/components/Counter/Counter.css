.Counter {
    display: flex;
    flex-flow: column nowrap;
}

.logoContainer {
    background-color: white;
}

.anotherContainer {
    width: 300px;
    height: 400px;
    background-color: blue;
    background-image: url(./metrica.svg);
}
