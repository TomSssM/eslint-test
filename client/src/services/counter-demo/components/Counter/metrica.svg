<svg width="71" height="34" viewBox="0 0 71 34" xmlns="http://www.w3.org/2000/svg"><path d="M19.958 18.24c.042 2.199.349 3.873.92 5.025.57 1.152 1.437 1.728 2.599 1.728.528 0 .993-.085 1.395-.254.401-.169.74-.37 1.014-.602.317-.232.592-.518.824-.856l.476 1.205c-.254.359-.56.665-.92.919a5.067 5.067 0 0 1-1.172.618c-.465.18-1.015.27-1.649.27-1.69 0-2.937-.73-3.74-2.188-.803-1.458-1.205-3.497-1.205-6.118 0-1.31.116-2.478.349-3.503.232-1.025.554-1.896.967-2.615.412-.718.903-1.263 1.474-1.632a3.37 3.37 0 0 1 1.87-.555c.697 0 1.32.18 1.87.539.55.359 1.004.898 1.363 1.616.36.719.613 1.612.761 2.679.148 1.067.159 2.309.032 3.725h-7.228zm1.03-5.753c-.602 1.046-.935 2.573-.998 4.58h5.801c.127-2.134-.08-3.692-.618-4.675-.54-.983-1.21-1.474-2.013-1.474-.845 0-1.57.523-2.172 1.57zM34.73 25.785c-.169.106-.37.19-.602.254a3.3 3.3 0 0 1-.856.095c-.95 0-1.653-.312-2.108-.935-.454-.623-.681-1.432-.681-2.425V11.172h-1.617V9.999h1.617l.158-3.17h1.332v3.17h2.726v1.173h-2.726v11.443c0 .803.121 1.385.364 1.744.243.36.608.539 1.094.539.232 0 .428-.027.586-.08.159-.052.29-.1.396-.142l.317 1.11zm8.064-14.328c-1.796.063-3.064 1.205-3.804 3.424v11.095H37.5V9.999h1.363l.063 3.201h.096c.126-.38.3-.76.523-1.14.222-.381.491-.725.808-1.031a4.22 4.22 0 0 1 1.094-.76 3.8 3.8 0 0 1 1.347-.365v1.553zm2.176-4.882a.987.987 0 0 1-.286-.729c0-.275.095-.518.286-.729a.937.937 0 0 1 .729-.317c.274 0 .512.106.713.317.2.211.301.454.301.73a.96.96 0 0 1-.301.728c-.2.19-.439.285-.713.285a.987.987 0 0 1-.73-.285zM46.428 10v15.977h-1.49V9.999h1.49zm9.367 14.74c.391-.169.724-.37.999-.602.317-.232.591-.518.824-.856l.475 1.236a3.542 3.542 0 0 1-.887.888c-.296.232-.682.439-1.157.618-.476.18-1.02.27-1.633.27-1.606 0-2.848-.724-3.725-2.172-.877-1.447-1.315-3.482-1.315-6.102 0-2.642.433-4.692 1.3-6.15.866-1.458 2.123-2.187 3.772-2.187.824 0 1.521.142 2.092.428.57.285 1.036.639 1.395 1.062l-.412 1.585a5.7 5.7 0 0 0-1.3-1.268c-.507-.36-1.099-.54-1.775-.54-1.226 0-2.13.614-2.71 1.84-.582 1.225-.872 2.969-.872 5.23 0 2.198.285 3.91.856 5.135.57 1.226 1.479 1.839 2.726 1.839.507 0 .956-.085 1.347-.254zm9.732-.57c.518-.613.925-1.353 1.22-2.22v-4.406h-1.711c-1.332 0-2.314.317-2.948.951-.635.634-.951 1.649-.951 3.044 0 1.14.227 2.018.681 2.63.454.614 1.062.92 1.823.92.74 0 1.368-.306 1.886-.92zm1.664 1.807a12.797 12.797 0 0 1-.206-.809 9.666 9.666 0 0 1-.174-1.347c-.38.76-.856 1.363-1.427 1.807-.57.444-1.257.666-2.06.666-1.078 0-1.955-.407-2.631-1.22-.677-.814-1.015-1.95-1.015-3.409 0-1.711.455-3.006 1.363-3.883.91-.877 2.23-1.315 3.963-1.315h1.743v-1.998c0-1.225-.221-2.113-.665-2.662-.444-.55-1.11-.825-1.997-.825a4.08 4.08 0 0 0-1.902.46 3.967 3.967 0 0 0-1.459 1.284l-.412-1.427c.317-.422.798-.797 1.443-1.125.644-.327 1.42-.491 2.33-.491 1.331 0 2.35.375 3.059 1.125.708.75 1.062 1.918 1.062 3.503v8.717c0 .55.031 1.105.095 1.665.063.56.148.988.253 1.284h-1.363zM5.735 9.995c-.317-1.172-.57-2.187-.76-3.232H4.88c.063 1.077.095 2.218.095 3.359V26H3.58V3.34h1.838l3.835 14.483c.158.602.38 1.775.475 2.63h.095c.127-.855.381-2.06.539-2.63L14.229 3.34h1.775V26h-1.522V10.122c0-1.078.064-2.218.127-3.359h-.127a79.774 79.774 0 0 1-.76 3.137l-3.391 12.614H9.094L5.735 9.995z" fill="red" fill-rule="evenodd"/></svg>
