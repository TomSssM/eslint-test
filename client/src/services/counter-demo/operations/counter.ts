import { QueryOperation } from '@client-libs/graphql-operations/decorators';
import { CounterEntity } from '@shared/metrika-dto/entities/counter';
import { CounterQueryArgs } from '@shared/metrika-dto/args/counter';
import { GraphQLEntityResultOperation } from '@client-libs/graphql-operations/operation';

@QueryOperation('counter', CounterEntity, CounterQueryArgs)
export class CounterRequest extends GraphQLEntityResultOperation<CounterEntity, CounterQueryArgs> {}
