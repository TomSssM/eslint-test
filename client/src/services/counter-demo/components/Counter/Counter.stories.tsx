import { CounterEntity } from '@shared/metrika-dto/entities/counter';
import React, { FC } from 'react';
import { Counter } from "./Counter";

export default {
    title: 'Counter',
};


export const Primary: FC = () => (
    <Counter counter={{
        id: '1',
        name: 'счетчик',
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    } as any as CounterEntity} />
);

Primary.args = {};
Primary.parameters = {};
