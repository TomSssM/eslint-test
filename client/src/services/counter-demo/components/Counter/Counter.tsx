import React from 'react';
import { CounterEntity } from '@shared/metrika-dto/entities/counter';
import styles from './Counter.css';
import MetricaLogo from './metrica.svg';

export interface CounterProps {
    counter: CounterEntity;
}

export function Counter({ counter: { id, name } }: CounterProps): React.ReactElement {
    return (
        <div className={styles.Counter}>
            <div className={styles.logoContainer}>
                <MetricaLogo />
            </div>
            <div className={styles.anotherContainer} />
            <span>{name}</span>
            <span>({id})</span>
        </div>
    );
}
