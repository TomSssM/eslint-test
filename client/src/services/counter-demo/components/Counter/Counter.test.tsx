import React from 'react';
import renderer from 'react-test-renderer';
import { CounterEntity } from '@shared/metrika-dto/entities/counter';
import { Counter } from './Counter';

describe('Counter', () => {
    test('should render', () => {
        const component = renderer.create(
            <Counter counter={{ id: '123', name: 'test' } as CounterEntity} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
