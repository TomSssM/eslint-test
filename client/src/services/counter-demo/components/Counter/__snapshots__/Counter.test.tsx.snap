// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render 1`] = `
<div>
  <div>
    <SvgrURL />
  </div>
  <div />
  <span>
    test
  </span>
  <span>
    (
    123
    )
  </span>
</div>
`;
